src/WebJobs.Script/Binding/Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -125,20 +131,60 @@ public Type TriggerValueType
                     throw new NotSupportedException("An HttpRequestMessage is required");
                 }
 
+                HttpResponseMessage responseObjectInFunction = null;
+                if (_isProxy)
```

Comment:
Let's move all of this implementation out of this method into helpers in a proxy util class